src/Employer/Employer.Web/Orchestrators/Part1/LocationOrchestrator.cs

```diff
@@ -68,58 +74,92 @@ public class LocationOrchestrator : EntityValidatingOrchestrator<Vacancy, Locati
             }
             else
             {
-                selectedOrganisation = 
+                selectedOrganisation =
                     editVacancyInfo.LegalEntities.Single(l => l.LegalEntityId == vacancy.LegalEntityId);
             }
 
+            var employerProfile =
+                await _recruitVacancyClient.GetEmployerProfileAsync(vacancy.EmployerAccountId, selectedOrganisation.LegalEntityId);
+
+            var newLocation = 
+                locationEditModel.SelectedLocation == LocationViewModel.UseOtherLocationConst 
+                ? locationEditModel.OtherLocationString 
+                : locationEditModel.SelectedLocation;
+
+            var matchingAddress = GetMatchingAddress(newLocation, employerProfile, selectedOrganisation);
 
-            if (m.UseOtherLocation == true)
+            if (matchingAddress != null)
             {
-                vacancy.EmployerLocation = new Vacancies.Client.Domain.Entities.Address
-                {
-                    AddressLine1 = m.AddressLine1,
-                    AddressLine2 = m.AddressLine2,
-                    AddressLine3 = m.AddressLine3,
-                    AddressLine4 = m.AddressLine4,
-                    Postcode = m.Postcode.AsPostcode()
-                };
+                vacancy.EmployerLocation = matchingAddress;
             }
             else
             {
-                vacancy.EmployerLocation = new Vacancies.Client.Domain.Entities.Address
+                vacancy.EmployerLocation = new Address 
                 {
-                    AddressLine1 = selectedOrganisation.Address.AddressLine1,
-                    AddressLine2 = selectedOrganisation.Address.AddressLine2,
-                    AddressLine3 = selectedOrganisation.Address.AddressLine3,
-                    AddressLine4 = selectedOrganisation.Address.AddressLine4,
-                    Postcode = selectedOrganisation.Address.Postcode.AsPostcode()
+                    AddressLine1 = locationEditModel.AddressLine1,
+                    AddressLine2 = locationEditModel.AddressLine2,
+                    AddressLine3 = locationEditModel.AddressLine3,
+                    AddressLine4 = locationEditModel.AddressLine4,
+                    Postcode = locationEditModel.Postcode.AsPostcode()
                 };
             }
-
+            
             return await ValidateAndExecute(
-                vacancy, 
+                vacancy,
                 v => _recruitVacancyClient.Validate(v, ValidationRules),
-                async v => 
+                async v =>
```

Comment:
You could move this bit into its own function

Reply:
fixed